### PR TITLE
feat: token deposit

### PR DESCRIPTION
**Pull request by obiyankenobi**
Added a message about token deposit in the create and mint tokens screens.

image

image

### CODE REVIEW

File: src/components/tokens/TokenAction.js [hunk 1]
```diff
@@ -134,6 +134,7 @@ class TokenAction extends React.Component {
     return (
       <div key={this.props.action}>
         <h2>{this.props.title}</h2>
+        <p>{this.props.subtitle}</p>
```
Comment (pedroferreira1) [inline src/components/tokens/TokenAction.js]: This will add a blank height when there is no `subtitle`. `<p>` tags have `margin-bottom`.

**Comment by pedroferreira1** [comment]
I think that we also should show the amount value that will be deposited in both screens (mint and create token), not only that will be 1%.

When the user changes the amount, it should show the HTR amount required.

File: src/screens/CreateToken.js [hunk 2]
```diff
@@ -249,6 +196,7 @@ class CreateToken extends React.Component {
         <p className="mt-5">Here you will create a new customized token. After the creation, you will be able to send this new token to other addresses.</p>
         <p>Custom tokens share the address space with all other tokens, including HTR. This means that you can send and receive tokens using any valid address.</p>
         <p>Remember to make a backup of your new token's configuration string. You will need to send it to other people to allow them to use your new token.</p>
+        <p>When creating and minting tokens, a <strong>deposit of {hathorLib.tokens.depositPercentage * 100}%</strong> in HTR is required. This will be returned when these tokens are later melted.</p>
```
Comment (msbrogli) [inline src/screens/CreateToken.js]: I would say: "... These will be returned if these tokens are melted later. For further information, see [link]".
I'm not sure about what link to use. Should we point to the RFC? Maybe we should have a document in our website.

File: src/components/tokens/TokenAction.js (hunk 1 above)
Comment (msbrogli) [inline src/components/tokens/TokenAction.js]: I think we should calculate the deposit in real-time and show to the user.
For example:
It the user typed 120 we would write "A deposit of 1.20 HTR will be made".

Comment (obiyankenobi) [reply src/components/tokens/TokenAction.js]: This is not a problem due to margin collapsing. `<p> margin-bottom` is 'ignored'.

`<p>`
image
`<form>`
image

File: src/screens/CreateToken.js (hunk 2 above)
Comment (obiyankenobi) [reply src/screens/CreateToken.js]: "When creating and minting tokens, a deposit of 1% in HTR is required. If these tokens are later melted, this HTR deposit will be returned. Read more about it here."

Linked to the RFC as we have no other document yet.

File: src/components/tokens/TokenAction.js (hunk 1 above)
Comment (obiyankenobi) [reply src/components/tokens/TokenAction.js]: Did that and updated the pictures. Can you review again?

Comment (msbrogli) [reply src/components/tokens/TokenAction.js]: Great. Thanks!

File: src/screens/CreateToken.js [hunk 3]
```diff
@@ -151,23 +107,22 @@ class CreateToken extends React.Component {
...
-        reject(message);
-      });
-    });
 
-    promise.then((token) => {
+    const retPromise = hathorLib.tokens.createToken(
+      address,
+      this.refs.shortName.value,
+      this.refs.symbol.value,
+      parseInt(this.state.amount*(10**hathorLib.constants.DECIMAL_PLACES), 10),
```
Comment (msbrogli) [inline src/screens/CreateToken.js]: We should move it to an utils method `valueFloatToInt(this.state.amount, hathorLib.constants.DECIMAL_PLACES)`.

File: src/utils/tokens.js [hunk 4]
```diff
@@ -75,6 +75,24 @@ const tokens = {
...
+   * Returns the deposit amount in 'pretty' format
+   *
+   * @param {number} mintAmount Amount of tokens to mint
+   *
+   * @memberof Tokens
+   * @inner
+   */
+  getDepositAmount(mintAmount) {
+    if (mintAmount) {
+      const amountValue = mintAmount*(10**hathorLib.constants.DECIMAL_PLACES);
```
Comment (msbrogli) [inline src/utils/tokens.js]: Use the suggested util method `valueFloatToInt(...)`.

File: src/screens/CreateToken.js (hunk 3 above)
Comment (obiyankenobi) [reply src/screens/CreateToken.js]: Done

File: src/utils/tokens.js (hunk 4 above)
Comment (obiyankenobi) [reply src/utils/tokens.js]: Done

**Comment by obiyankenobi** [comment]
Waiting for new lib version so we can update package.json

**Pull request merged**